src/rpc/misc.cpp

```diff
@@ -158,11 +158,8 @@ UniValue validateaddress(const JSONRPCRequest& request)
 
 #ifdef ENABLE_WALLET
     CWallet * const pwallet = GetWalletForJSONRPCRequest(request);
-
-    LOCK2(cs_main, pwallet ? &pwallet->cs_wallet : nullptr);
-#else
-    LOCK(cs_main);
 #endif
+    LOCK(cs_main);
```

Comment:
How about not changing behaviour:
```cpp
LOCK(cs_main);
#ifdef ENABLE_WALLET
if (pwallet) LOCK(pwallet->cs_wallet);
#endif
```
As it is, the wallet can change between the `cs_main` lock and `cs_wallet` lock. This way the diff is also shorter.

Reply:
@promag Regarding your edit - do you mean that you ran into Clang thread safety analysis warnings?